JenifarFathimaH/00 - Projects/04 - Frontend Projects/E-Commerce Project/Assets/JS/Customer/CustomerSignup.js

```diff
@@ -0,0 +1,50 @@
+const USER_NAME = document.getElementById("inp-signupUserName");
+const EMAIL = document.getElementById("inp-signupEmail");
+const AGE = document.getElementById("inp-signupAge");
+const PHONE = document.getElementById("inp-signupPhone");
+const NEW_PASSWORD = document.getElementById("inp-newSignupPassword");
+const CONFORM_PASSWORD = document.getElementById("inp-conformSignupPassword");
+
+let CUSTOMER_LIST = JSON.parse(localStorage.getItem("customer_list")) || [];
+
+let customerListObj = {};
+
+let SignUpMethods = {};
+
+SignUpMethods.SignupUser = function () {
+
+    //object values
+    customerListObj.strUsername = USER_NAME.value;
+    customerListObj.strEmail = EMAIL.value;
+    customerListObj.nAge = AGE.value;
+    customerListObj.nPhone = PHONE.value;
+    customerListObj.strNewPassword = NEW_PASSWORD.value;
+    customerListObj.strConformPassword = CONFORM_PASSWORD.value;
+    
+
+    if (NEW_PASSWORD.value != CONFORM_PASSWORD.value){
+        alert("Password Mismatch!!")
+        return;
+    }
+
+    if (CUSTOMER_LIST.find(customer => customer.strUsername === USER_NAME.value || customer.strEmail === EMAIL.value)) {
+        alert("Username or Email already exists!!");
+        return;
+    }
+    SignUpMethods.SignUpCheck();
+}
+
+SignUpMethods.SignUpCheck = function () {
+    [USER_NAME, EMAIL, AGE, PHONE, NEW_PASSWORD, CONFORM_PASSWORD].forEach(input => input.value = "");
```

Comment:
Different approach. Looks good! 😊

Another possible approach is **form.reset()**

Reply:
Done